src/components/Campaign.tsx

```diff
@@ -26,18 +26,20 @@ export const Campaign = () => {
   } = useAppServices();
 
   const [searchParams] = useSearchParams();
-
   const campaignContentQuery = useGetCampaignContent(idCampaign);
   const campaignContentMutation = useUpdateCampaignContent();
-  const { save } = useSingletonEditor({
-    initialContent: campaignContentQuery.data,
-    onSave: async (content: Content) => {
-      campaignContentMutation.mutate({ idCampaign, content });
-    },
-  });
-
   const intl = useIntl();
 
+  const { save } = useSingletonEditor(
+    {
+      initialContent: campaignContentQuery.data,
+      onSave: (content: Content) => {
+        campaignContentMutation.mutate({ idCampaign, content });
+      },
+    },
+    [campaignContentQuery.data, idCampaign]
```

Comment:
I know that it does not change, but I would add the dependency to `campaignContentMutation.mutate` in order to make it clear. 

If we would have the lint rule for our `useSingletonEditor` hook, it would require it.

```suggestion
    [campaignContentQuery.data, idCampaign, campaignContentMutation.mutate]
```